botorch/models/ensemble.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Ensemble Models: Simple wrappers that allow the usage of ensembles
+via the BoTorch Model and Posterior APIs.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from botorch.acquisition.objective import PosteriorTransform
+from botorch.exceptions.errors import UnsupportedError
+from botorch.models.model import Model
+from botorch.posteriors.ensemble import EnsemblePosterior
+from torch import Tensor
+
+
+class EnsembleModel(Model, ABC):
+    r"""
+    Abstract base class for ensemble models.
+
+    :meta private:
+    """
+
+    @abstractmethod
+    def forward(self, X: Tensor) -> Tensor:
+        r"""Compute the (ensemble) model output at X.
+
+        Args:
+            X: A `batch_shape x n x d`-dim input tensor `X`.
+
+        Returns:
+            A `batch_shape x s x n x m`-dimensional output tensor where
+            `s` is the size of the ensemble.
+        """
+        pass  # pragma: no cover
+
+    def forward_(self, X: Tensor) -> Tensor:
+        return self.forward(X=X)
```

Comment:
Let's maybe call this `_forward` to keep it private. Also, if we agree that we want to phase out `DeterminsticModel` and instead just always use `EnsembleModel`s with ensemble size 1 (a question we still need to answer - @saitcakmak, @dme65, @sdaulton any thoughts here), then we should probably add a TODO to remove this method once `DeterminsticModel` is gone.

Reply:
Done, I changed it according to your suggestion.